validate/error.go

```diff
@@ -0,0 +1,91 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ComplianceLevel represents the OCI compliance levels
+type ComplianceLevel int
+
+const (
+	// MAY-level
+	ComplianceMay ComplianceLevel = iota
+	ComplianceOptional
+	// SHOULD-level
+	ComplianceShould
+	ComplianceShouldNot
+	ComplianceRecommended
+	ComplianceNotRecommended
+	// MUST-level
+	ComplianceMust
+	ComplianceMustNot
+	ComplianceShall
+	ComplianceShallNot
+	ComplianceRequired
+)
+
+// ErrorCode represents the compliance content
+type ErrorCode int
+
+const (
+	DefaultFilesystems ErrorCode = iota
+)
+
+// Error represents an error with compliance level and OCI reference
+type Error struct {
+	Level     ComplianceLevel
+	Reference string
+	Err       error
+}
+
+//FIXME: change to tagged spec releases
+const referencePrefix = "https://github.com/opencontainers/runtime-spec/blob/master/"
+
+var ociErrors = map[ErrorCode]Error{
+	DefaultFilesystems: Error{Level: ComplianceShould, Reference: "config-linux.md#default-filesystems"},
+}
+
+// ParseLevel takes a string level and returns the OCI compliance level constant
+func ParseLevel(level string) ComplianceLevel {
+	switch strings.ToUpper(level) {
+	case "MAY":
+		fallthrough
+	case "OPTIONAL":
+		return ComplianceMay
+	case "SHOULD":
+		fallthrough
+	case "SHOULDNOT":
+		fallthrough
+	case "RECOMMENDED":
+		fallthrough
+	case "NOTRECOMMENDED":
+		return ComplianceShould
+	case "MUST":
+		fallthrough
+	case "MUSTNOT":
+		fallthrough
+	case "SHALL":
+		fallthrough
+	case "SHALLNOT":
+		fallthrough
+	case "REQUIRED":
+		return ComplianceMust
+	default:
+		return ComplianceMust
```

Comment:
I think the default should be to raise an “unrecognized level” error.  Silently defaulting to MUST seems confusing.